### ISSUE TITLE
getSessionData ends before all files are in

### ISSUE THREAD

**Issue by harunsmrkovic** [digime/digime-sdk-ios#135]
```
 LOG  Retrieved 18_3_32_7_201_D201805_1.json 6
 LOG  Retrieved 18_3_32_7_201_D201804_1.json 6
 LOG  Retrieved 18_3_32_7_201_D201806_1.json 6
 LOG  Retrieved 18_3_32_7_201_D201811_1.json 6
 WARN  [digi.me] Get Files Stream Ended with a warning [Error: The data couldn’t be read because it isn’t in the correct format.]
 LOG  Marking fetchData as done
 LOG  Retrieved 18_5_31_10_2_D201809_1.json 10
 LOG  Retrieved 18_1_1_2_1_D201904_1.json 3
 LOG  Retrieved 18_1_4_3_1_D201902_1.json 41
```

Log above suggests that the completion block of `getSessionDataWithDownloadHandler` gets called with an error before the streaming of files has actually ended, as events are being dispatched about incoming files.

I've seen a similar report in Android SDK Github - maybe it is related.

If you cannot reproduce it at all, let me know and I can debug it bit more, maybe it is in our bridge...

**Comment by brokenzz** [comment]
Hi Harun,

Please provide:
- a code snippet reponsible for producing  above log
- information on whether the session data fetch completion handler is executed with or without an error object.

**Comment by brokenzz** [comment]
Thanks, we'll investigate this.

**Comment by brokenzz** [comment]
Quick update - following investigation we identified a scenario where session completion could be called prematurely, specifically if an error had occurred during `fileList` fetching. Handling of this case has been improved. For details, see #139.

This will be included in the next version of the SDK (3.0.1). You will not need to do anything other than install the latest version.
